transports/quic/src/muxer.rs

```diff
@@ -89,6 +143,93 @@ impl QuicMuxer {
         }
     }
 }
+impl StreamMuxer for QuicMuxer {
+    type Substream = Substream;
+    type Error = Error;
+
+    fn poll_address_change(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<libp2p_core::Multiaddr, Self::Error>> {
+        self.inner.lock().poll_connection(cx);
+        // TODO
+        Poll::Pending
+    }
+
+    fn poll_inbound(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Self::Substream, Self::Error>> {
+        let mut inner = self.inner.lock();
+        inner.poll_connection(cx);
+        if let Some(substream_id) = inner.connection.pop_incoming_substream() {
+            inner.substreams.insert(substream_id, Default::default());
+            let substream = Substream::new(substream_id, self.inner.clone());
+            Poll::Ready(Ok(substream))
+        } else {
+            inner.poll_event_waker = Some(cx.waker().clone());
+            Poll::Pending
+        }
```

Comment:
What do you think of:

```suggestion
        let substream_id = match inner.connection.pop_incoming_substream() {
        	Poll::Ready(id) => id,
        	Poll::Pending => {
        		inner.poll_event_waker = Some(cx.waker().clone());
        		return Poll::Pending;
        	}
        };

        inner.substreams.insert(substream_id, Default::default());
        let substream = Substream::new(substream_id, self.inner.clone());
        
        Poll::Ready(Ok(substream))
````

Reply:
This coding style would visually separate the two paths, which IMO makes it easier to see at a glance, what is happening.

Reply:
Another idea: We could make a function on `inner` to say `poll_incoming_substream_id`. Then we could use the `ready!` macro here because the waker would be registered inside, not sure if the indirection is worth it.

Reply:
Went with the first suggestion, which imo is easier to read than splitting it into to methods.